runtime/lua/vim/shared.lua

```diff
@@ -987,4 +987,35 @@ do
   end
 end
 
+-- TODO(lewis6991): Private for now until:
+-- - There are other places in the codebase that could benefit from this
+--   (e.g. LSP), but might require other changes to accommodate.
+-- - Invalidation of the cache needs to be controllable. Using weak tables
+--   is an acceptable invalidation policy, but it shouldn't be the only
+--   one.
+-- - I don't think the story around `hash_fn` is completely thought out. We
+--   may be able to have a good default hash_fn by hashing each argument,
+--   so basically a better 'concat'.
+--
+--- Memoizes a function {fn} using {hash_fn} to hash the arguments.
+--- Internally uses a weak table to cache the results of {fn}.
```

Comment:
Specifically: mention here that this means the cache will be invalidated by Lua(JIT) GC, which means it will expire automatically (but uncontrollably)?

Reply:
Not a LuaJIT specific thing, but sure.